server/coordinator/watch/watch.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package coordinator
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/storage"
+	"github.com/pkg/errors"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+type ShardEventType uint64
+
+const (
+	shardPath                  = "shards"
+	EventDelete ShardEventType = 0
+	EventPut                   = 1
+)
+
+// ShardWatch used to watch the distributed lock of shard, and provide the corresponding callback function.
+type ShardWatch struct {
+	rootPath       string
+	etcdClient     *clientv3.Client
+	eventCallbacks []func(shardID storage.ShardID, nodeName string, eventType ShardEventType) error
```

Comment:
Maybe we should define an interface for the callbacks for explicit meaning of its parameter:
```
type ShardEventCallback interface {
   OnShardRegistered(ShardRegisterEvent) error
   OnShardExpired(ShardExpireEvent) error
}
```